### PR TITLE
refactor: refactor move_primary in greedy_load_balancer

### PR DESCRIPTION
**Pull request by levy5307**

### CODE REVIEW

**Review by Smityz — APPROVED** [review]

File: src/meta/greedy_load_balancer.cpp [hunk 1]
```diff
@@ -784,105 +784,112 @@ void ford_fulkerson::update_flow(int pos,
...
     }
-    return true;
+
+    dassert_f(selected != potential_moving.end(),
+              "can't find gpid to move from({}) to({})",
+              from.to_string(),
+              to.to_string());
+    auto res = *selected;
+    potential_moving.erase(selected);
+    return res;
```
Comment (Shuo-Jia) [inline src/meta/greedy_load_balancer.cpp:892]: directly return `*selected`?

Comment (levy5307) [reply src/meta/greedy_load_balancer.cpp:892]: We can't do it like this, because selected will be removed by `potential_moving.erase(selected);`

Comment (Shuo-Jia) [reply src/meta/greedy_load_balancer.cpp:892]: So the remove should't put here, the function only `select`, better add remove logic, better put out

Comment (levy5307) [reply src/meta/greedy_load_balancer.cpp:892]: Yes, you are right. But in set of cpp, it doesn't support to move *selected from potential_moving. It only support remove iter from it.

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**